pkg/cmd/server/origin/run_components.go

```diff
@@ -167,7 +168,12 @@ func (c *MasterConfig) RunDNSServer() {
 	}
 
 	go func() {
-		err := dns.ListenAndServe(config, c.DNSServerClient(), c.EtcdClient)
+		etcdClient, err := etcd.GetAndTestEtcdClient(c.Options.EtcdClientInfo)
```

Comment:
I'm actually wondering why we do this at all... @smarterclayton, is there a reason we're not giving skynet a nil etcdclient?